Standards/scs-0100-v4-flavor-naming.md

```diff
@@ -0,0 +1,549 @@
+
+While flavors can be discovered (`openstack flavor list`), it is helpful for users (DevOps teams),
+to have a naming scheme that is used across all SCS flavors, so flavor names have the same meaning everywhere.
+
+While not all details will be encoded in the name, the key features should be obvious:
+Number of vCPUs, RAM, Root Disk.
+Extra features are important as well: There will be flavors with GPU support, fast disks for databases,
+memory-heavy applications, and other useful aspects of an instance.
+
+It may also be important to make the CPU generation clearly recognizable, as this is always a topic in
+discussions with customers.
+
+Note that not all relevant properties of flavors can be discovered; creating a specification
+to address this is a separate but related effort to the name standardization.
+Commonly used infrastructure-as-code tools do not provide a way to use discoverability
+features to express something like "I want a flavor with 2 vCPUs, 8GiB of RAM, a local
+20GB SSD disk and Infiniband support but I don't care whether it's AMD or intel" in a
+reasonable manner. Using flavor names to express this will thus continue to be useful
+and we don't expect the need for standardization of flavor names to go away until
+the commonly used IaC tools work on a higher abstraction layer than they currently do.
+
+## Design Considerations
+
+### Type of information included
+
+From discussions of our operators with their customers we learned that
+the following characteristics are important in a flavor description:
+
+| Type              | Description                                            |
+| :---------------- | :----------------------------------------------------- |
+| Generation        | CPU Generation                                         |
+| Number of CPU     | Number of vCPUs - suffixed by L,V,T,C (see below)      |
+| Amount of RAM     | Amount of memory available for the VM                  |
+| Performance Class | Ability to label high-performance CPUs, disks, network |
+| CPU Type          | X86-intel, X86-amd, ARM, RISC-V, Generic               |
+| "bms"             | Bare Metal System (no virtualization/hypervisor)       |
+
+This list is likely not comprehensive and will grow over time.
+
+Rather than using random names `s5a.medium` and assigning a discrete set of properties
+to them, we wanted to come up with a scheme that allows to systematically derive
+names from properties and vice versa. The scheme allows for short names (by not
+encoding all details) as well as very detailed longer names.
+
+## Complete Proposal for systematic flavor naming
+
+| Prefix | CPUs & Suffix     | RAM[GiB]           | optional: Disk[GB]&type       | opt: extensions |
+| ------ | ----------------- | ------------------ | ----------------------------- | ----------------|
+| `SCS-` | N`L/V/T/C`\[`i`\] | `-`N\[`u`\]\[`o`\] | \[`-`\[M`x`\]N\[`n/h/s/p`\]\] | \[`_`EXT\]      |
```

Comment:
Syntactically, I believe using `|` as an OR operator is more common than `/`.
Furthermore, I think these operators should not be in backticks - if my assumption is correct that the monospace characters are supposed to be literals whereas everything else outside the backticks is either a placeholder (such as M or N) or an operator (such as [ ] for optionals).

```suggestion
| Prefix | CPUs & Suffix              | RAM[GiB]           | opt: Disk[GB]&type                     | opt: extensions |
| ------ | -------------------------- | ------------------ | -------------------------------------- | ----------------|
| `SCS-` | N`L`\|`V`\|`T`\|`C`\[`i`\] | `-`N\[`u`\]\[`o`\] | \[`-`\[M`x`\]N\[`n`\|`h`\|`s`\|`p`\]\] | \[`_`EXT\]      |
```

Reply:
I will repeat myself:
> It's an issue of its own [...], and that issue then should fix the other versions of the standard as well.

Reply:
Whoopsie, ignore my comment - I completely missed the scope of this PR - my bad.

Reply:
In other words: can you please only review things that are new in v4? As of now, IIRC, there aren't any, so you're done...

Reply:
> In other words: can you please only review things that are new in v4? As of now, IIRC, there aren't any, so you're done...

Yes, as I stated: my bad. I completely mistook this PR for something else because I was misled by the PR diff showing a whole new file. Sorry.